modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalImportListTypeDefinition.tsx

```diff
@@ -236,34 +242,34 @@ export default function ModalImportListTypeDefinition({
 						name={listTypeDefinitionJSONInputId}
 						onChange={({target}) => {
 							const inputFile = target.files?.item(0);
+							if (inputFile) {
+								setFile({
+									fileName: inputFile?.name,
+									inputFile,
+								});
+
+								const fileReader = new FileReader();
+
+								fileReader.onload = () => {
+									try {
+										const objectDefinitionJSON = JSON.parse(
+											fileReader.result as string
+										) as {externalReferenceCode: string};
+
+										setExternalReferenceCode(
+											objectDefinitionJSON.externalReferenceCode
+										);
+									}
+									catch (error) {
+										setExternalReferenceCode('');
+									}
+								};
 
-							setFile({
-								fileName: inputFile?.name,
-								inputFile,
-								inputFileValue: target.value,
-							});
-
-							const fileReader = new FileReader();
-
-							fileReader.onload = () => {
-								try {
-									const objectDefinitionJSON = JSON.parse(
-										fileReader.result as string
-									) as {externalReferenceCode: string};
-
-									setExternalReferenceCode(
-										objectDefinitionJSON.externalReferenceCode
-									);
-								}
-								catch (error) {
-									setExternalReferenceCode('');
-								}
-							};
-							fileReader.readAsText(inputFile!);
+								fileReader.readAsText(inputFile!);
```

Comment:
Hey @larissacribeiro, could you move this `fileReader.readAsText(`) right below the `fileReader` declaration on line `251`?

Also, since you already made the condition to check if the `inputFile` exists on line `245`, you don't need to put that `!` in the parameter, you can remove it.

Reply:
Hi @danvb112, just updated it, thanks!